src/components/calcite-switch/calcite-switch.tsx

```diff
@@ -60,109 +134,76 @@ export class CalciteSwitch {
+      (!this.disabled && this.el.closest("label") && e.target === this.inputEl) ||
       (!this.el.closest("label") && e.target === this.el)
     ) {
-      this.updateSwitch(e);
+      this.updateSwitch();
     }
   }
 
-  @Listen("keydown") keyDownHandler(e: KeyboardEvent): void {
+  @Listen("keydown")
+  keyDownHandler(e: KeyboardEvent): void {
     const key = getKey(e.key);
     if (!this.disabled && (key === " " || key === "Enter")) {
-      this.updateSwitch(e);
+      this.updateSwitch();
     }
   }
 
-  @Watch("switched") switchWatcher(): void {
-    this.switched
-      ? this.inputProxy.setAttribute("checked", "")
-      : this.inputProxy.removeAttribute("checked");
-  }
-
-  private inputProxy: HTMLInputElement;
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
 
   connectedCallback(): void {
-    this.setupProxyInput();
-  }
-
-  disconnectedCallback(): void {
-    this.observer.disconnect();
+    this.guid = this.el.id || `calcite-switch-${guid()}`;
+    this.renderInput();
   }
 
-  componentWillRender(): void {
-    this.syncProxyInputToThis();
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  private renderInput(): void {
+    this.inputEl = document.createElement("input");
+    this.switched && this.inputEl.setAttribute("checked", "");
+    this.inputEl.disabled = this.disabled;
+    this.inputEl.id = `${this.guid}-input`;
+    this.inputEl.name = this.name;
+    this.inputEl.style.setProperty("bottom", "0", "important");
+    this.inputEl.style.setProperty("left", "0", "important");
+    this.inputEl.style.setProperty("margin", "0", "important");
+    this.inputEl.style.setProperty("opacity", "0", "important");
+    this.inputEl.style.setProperty("outline", "none", "important");
+    this.inputEl.style.setProperty("padding", "0", "important");
+    this.inputEl.style.setProperty("position", "absolute", "important");
+    this.inputEl.style.setProperty("right", "0", "important");
+    this.inputEl.style.setProperty("top", "0", "important");
+    this.inputEl.style.setProperty("transform", "none", "important");
+    this.inputEl.style.setProperty("-webkit-appearance", "none", "important");
+    this.inputEl.style.setProperty("z-index", "-1", "important");
+    this.inputEl.type = "checkbox";
+    if (this.value) {
+      this.inputEl.value = this.value;
+    }
+    this.el.appendChild(this.inputEl);
   }
 
   render(): VNode {
     const dir = getElementDir(this.el);
-
     return (
-      <Host
-        aria-checked={this.switched.toString()}
-        dir={dir}
-        tabIndex={this.disabled ? -1 : this.tabIndex}
-      >
+      <Host aria-checked={this.switched.toString()} dir={dir} tabindex={this.disabled ? -1 : 0}>
```

Comment:
nitpick: can we add these props on an internal node so they are encapsulated instead of in light DOM?